syncer/blockssync/blocks.go

```diff
@@ -0,0 +1,49 @@
+package blockssync
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+type blockFetcher interface {
+	GetBlocks(context.Context, []types.BlockID) error
+}
+
+// Sync requests last specified blocks in background.
+func Sync(ctx context.Context, logger *zap.Logger, requests <-chan []types.BlockID, fetcher blockFetcher) error {
+	var (
+		eg     errgroup.Group
+		lastch = make(chan []types.BlockID)
+	)
+	eg.Go(func() error {
+		var (
+			send chan []types.BlockID
+			last []types.BlockID
+		)
+		for {
+			select {
+			case <-ctx.Done():
+				close(lastch)
+				return ctx.Err()
+			case req := <-requests:
```

Comment:
since the order is not deterministic, if this case got selected twice in a row, the previous batch will be ignored?

Reply:
what order is not deterministic?

Reply:
i will change it back to aggregate to make more robust